### PR TITLE
Working mech DLC: Clarke and KA

### PR DESCRIPTION
**Pull request by TheRyeGuyWhoWillNowDie**
All of this is tested, and looks to be working.

Currently, lavaland mechs are not very helpful. Yes, they ignore ash storms, can walk over lava, etc, but they lack the ability to do any real damage, and the firefighter (the current "real" mining mech) is slow as fuck. This PR kills two birds with one stone.

Clarke Mech
~~stolen~~ ported directly from tg, the clarke functions similarly to how it does there. It moves faster than the other working mechs, and extremely quickly on lavaland. This speed comes at the cost of the inability to strafe, moving costs more power. However, the clarke is able to shoot behind itself to compensate for being unable to strafe. This makes it a choice for players who favor speed over defense, as it has lower armor than the Firefighter, and cannot get upgraded using goliath plates. To be specific, clarke's armor is 
MELEE = 20, BULLET = 10, LASER = 20, ENERGY = 10, BOMB = 60, BIO = 0, RAD = 70, FIRE = 100, ACID = 100
which, while quite low, isn't much of an issue because clarke is fast and can simply avoid getting hit.
Clarke is unlocked with a technology requiring industrial engineering.
Clarke also has an internal ore box, and picks up ore like how a ripley with a clamp and ore box would. You can empty the ore box in the mech status menu. The ore box cannot be detached. The Clarke gives you a diagnostic HUD while you are inside of it, to tell miners how much HP their mech has left.

Exosuit PKA
Ported from TG as well.
Unlike last time, this exosuit PKA doesn't fire 100 damage kinetic forces. The projectiles fired from the mech PKA do 40 damage like the real PKA, however they travel 2 tiles further, and the mech can shoot substantially faster than a normal PKA can. Firing costs 30 power from the cell per shot.
It's unlocked with basic mining tech.

# Spriting
Clarke mecha:
image
Exosuit PKA:
image

# Wiki Documentation

See the first section of the PR. Put this information into the guide to robotics.

# Changelog

Fikou, InvalidArgument3, ported by TheRyeGuyWhoWillNowDie

:cl:  
rscadd: Clarke mech!
rscadd: Exosuit PKA
/:cl:

### CODE REVIEW

**Comment by tattax** [comment]
can you link the original prs

**Comment by TheRyeGuyWhoWillNowDie** [comment]
> can you link the original prs

most of https://github.com/tgstation/tgstation/pull/49563

some of https://github.com/tgstation/tgstation/pull/63809

and https://github.com/tgstation/tgstation/pull/60384

**Review by cuackles — APPROVED** [review]
yes

File: code/game/mecha/mech_fabricator.dm [hunk 1]
```diff
@@ -503,8 +504,11 @@
 		var/datum/design/D = SSresearch.techweb_design_by_id(v)
 		if(D.build_type & MECHFAB)
 			if(ispath(D.build_path, /obj/item/mecha_parts/mecha_equipment/weapon) && !combat_parts_allowed) // Yogs -- ID swiping for combat parts
-				if(D.build_path != /obj/item/mecha_parts/mecha_equipment/weapon/energy/plasma) // Yogs -- Special snowflake exception for mecha plasma cutters.
+				var/obj/item/mecha_parts/mecha_equipment/weapon/check = new D.build_path(src)
+				if(check.restricted)
```
Comment (Bibby0110) [inline code/game/mecha/mech_fabricator.dm]:
```suggestion
				if(initial(D.restricted))
```

File: code/game/mecha/mech_fabricator.dm [hunk 2]
```diff
@@ -503,8 +504,11 @@
...
+					qdel(check)
```
Comment (Bibby0110) [inline code/game/mecha/mech_fabricator.dm]:
```suggestion
```

File: code/game/mecha/mech_fabricator.dm [hunk 3]
```diff
@@ -503,8 +504,11 @@
...
 					continue
+				qdel(check)
```
Comment (Bibby0110) [inline code/game/mecha/mech_fabricator.dm]:
```suggestion
```

**Review by Bibby0110 — APPROVED** [review]

**Pull request merged**